docs/management-sdk-migration.md

```diff
@@ -0,0 +1,125 @@
+# `IServiceManager` to `ServiceManager` Migration Guidance
+
+This guidance is intended to assist in the migration to `ServiceManager` from `IServiceManager`. It will focus on side-by-side comparisons for similar operations between the legacy and the new APIs. The new APIs will be released in version 10.0.0 .
+
+We assume that you are familiar with the usage of `ServiceManagerBuilder` and the related APIs, otherwise, please refer the [Azure SignalR Service Management SDK ](management-sdk-guide.md) other than this guidance.
+- [`IServiceManager` to `ServiceManager` Migration Guidance](#iservicemanager-to-servicemanager-migration-guidance)
+  - [Migration Benefits](#migration-benefits)
+  - [Change Overview](#change-overview)
+  - [API comparisons](#api-comparisons)
+    - [Build service manager](#build-service-manager)
+    - [Create service hub context](#create-service-hub-context)
+    - [Negotiation](#negotiation)
+    - [Send Messages and Manage Groups](#send-messages-and-manage-groups)
+    - [Health check](#health-check)
+
+
+
+## Migration Benefits
+* The new APIs provide more functionalities to manage your clients and groups, such as closing a connection by connection id, checking if a connection exists, if a user exists, if a group exists. 
+* The new APIs provide more options for negotiation, such as whether the client is a diagnostic client.
+* The new APIs are more friendly for negotiation with multiple SignalR Service instances. `IServiceManager.GetClientEndpoint` and `IServiceManager.GenerateClientAccessToken` are combined into one method to make sure the client endpoint and the access token come from the same SignalR Service endpoint. An `HttpContext` instance is passed into the endpoint router to provide more information for the routing. 
+<!--Todo Add link about sharding doc-->
+
+## Change Overview
+Generally we use abstract classes to replace interfaces for better extensibility under different frameworks (.Net Standard 2.0, .Net Core 3.1, ...). `IServiceManager` is replaced by abstract class `ServiceManager`; `IServiceHubContext` is replaced by `ServiceHubContext`. What's more, negotiation is moved from `IServiceManager` to `ServiceHubContext`.
+## API comparisons
+### Build service manager
+
+**Legacy APIs**
+```cs
+var serviceManager = new ServiceManagerBuilder()
+    .WithOptions(o => o.ConnectionString = "<Your ConnectionString>")
+    .Build(); //build method is changed
+```
+
+**New APIs**
+```cs
+var serviceManager = new ServiceManagerBuilder()
+    .WithOptions(o => o.ConnectionString = "<Your ConnectionString>")
+    .WithLoggerFactory(loggerFactory) // optional, specify a custom logger factory instance for the whole management SDK
+    .BuildServiceManager(); // note this
+```
+
+<!--Add sharding link-->
+### Create service hub context
+
+**Legacy APIs**
+```cs
+var serviceHubContext = await serviceManager.CreateServiceHubContextAsync("<Your Hub Name>", loggerFactory, cancellationToken);
```

Comment:
question: L81 has a `CreateHubContextAsync`, here has a `CreateServiceHubContextAsync`, are these 2 different methods creating different objects?

Reply:
They are error. They should be both `CreateHubContextAsync`